src/views/studio/studio-admin-panel.jsx

```diff
@@ -0,0 +1,96 @@
+/* eslint-disable react/jsx-no-bind */
+import React, {useState, useEffect} from 'react';
+import {connect} from 'react-redux';
+import classNames from 'classnames';
+
+
+import {selectIsAdmin} from '../../redux/session.js';
+import {selectStudioId} from '../../redux/studio.js';
+
+import AdminPanel from '../../components/adminpanel/adminpanel.jsx';
+
+const adminPanelOpenKey = 'adminPanelToggled_studios';
+const adminPanelOpenClass = 'mod-view-admin-panel-open';
+
+/**
+ * Propagate the admin panel openness to localStorage and set a class name
+ * on the #view element.
+ * @param {boolean} value - whether the admin panel is now open.
+ */
+const storeAdminPanelOpen = value => {
+    try {
+        localStorage.setItem(adminPanelOpenKey, value ? 'open' : 'closed');
+    } catch (e) {
+        // eslint-disable-next-line no-console
+        console.error('Could not set adminPanelToggled_studios in local storage', e);
+    }
+    try {
+        document.querySelector('#view').classList
+            .toggle(adminPanelOpenClass, value);
```

Comment:
Yeah there was no machinery for doing this through the component and i was wary about adding state/callbacks to thread from this high up in the component tree. But I think mostly because this felt like such a special case. Definitely open to thoughts, I think I mostly started down this path because i thought #view was extra-react-ular, but I think I was thinking of the #app element.